components/script/dom/globalscope.rs

```diff
@@ -2180,42 +2184,46 @@ impl GlobalScope {
     pub fn add_uncaught_rejection(&self, rejection: HandleObject) {
         self.uncaught_rejections
             .borrow_mut()
-            .push(*Heap::boxed(rejection.get()));
+            .push(Heap::boxed(rejection.get()));
     }
 
     pub fn remove_uncaught_rejection(&self, rejection: HandleObject) {
         let mut uncaught_rejections = self.uncaught_rejections.borrow_mut();
 
         if let Some(index) = uncaught_rejections
             .iter()
-            .position(|promise| *promise == *Heap::boxed(rejection.get()))
+            .position(|promise| *promise == Heap::boxed(rejection.get()))
         {
             uncaught_rejections.remove(index);
         }
     }
 
-    pub fn get_uncaught_rejections(&self) -> &DomRefCell<Vec<Heap<*mut JSObject>>> {
+    #[allow(clippy::vec_box)]
+    // `Heap` values must stay boxed, as they need semantics like `Pin` ie they cannot be moved
+    pub fn get_uncaught_rejections(&self) -> &DomRefCell<Vec<Box<Heap<*mut JSObject>>>> {
         &self.uncaught_rejections
     }
 
     pub fn add_consumed_rejection(&self, rejection: HandleObject) {
         self.consumed_rejections
             .borrow_mut()
-            .push(*Heap::boxed(rejection.get()));
+            .push(Heap::boxed(rejection.get()));
```

Comment:
Can this be reverted?

Reply:
Absolutely.

Reply:
@sagudev ^^

Reply:
Am not sure I understand question. But given that we store Box<Heap<>> (and we must) we also need ` .push(Heap::boxed(rejection.get()))` (this is what we have now actually).

Reply:
@sagudev To rephrase, since we are using the `allow` directive to permit a `<Vec<Box<Heap<*mut JSObject>>>>` syntax, would a `.push(Heap::boxed(rejection.get()))` be the correct code?

Or do we need to include a `*` (like so: `.push(*Heap::boxed(rejection.get()))`)?

Reply:
Apologies, this looks fine. I was looking at the latest commit when I should have been looking at the diff from main. Please just move the comments. Otherwise this looks good.